src/alp/appcom/core.py

```diff
@@ -63,12 +77,37 @@ def fit(self, data, data_val, model=None, *args, **kwargs):
         if _recompile is True:
             self.model_dict = self.backend.to_dict_w_opt(self.model,
                                                          self.metrics)
-        self.res = self.backend.fit(copy.deepcopy(self.model_dict), data,
-                                    data_val, *args, **kwargs)
+        res = self.backend.fit(self.backend_name, self.backend_version,
+                               copy.deepcopy(self.model_dict), data,
+                               data_val, *args, **kwargs)
+        self.mod_id = res[0]
+        self.data_id = res[1]
+
         self.trained = True
 
         return self.res
 
+    def load_model(self, mod_id, data_id):
+        self.__model_dict['mod_id'] = mod_id
+        self.__model_dict['data_id'] = data_id
+        models = get_models()
+        model_db = models.find_one({'mod_id': mod_id, 'data_id': data_id})
+        self._switch_backend(model_db)
+        self.model_dict = model_db['model_arch']
+        self.trained = True
+
+    def _switch_backend(self, model_db):
+        if model_db['backend_name'] != self.backend_name:
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Access to member 'backend_name' before its definition line 102](https://www.codacy.com/app/thomas-boquet/python-alp/file/2492275084/issues/source?bid=3232749&fileBranchId=3265934#l100)